src/shogun/base/SGObject.h

```diff
@@ -966,6 +934,26 @@ class CSGObject
 	 */
 	void observe(const Some<ObservedValue> value);
 
+	/**
+	 * Observe a parameter value given some information
+	 * @tparam T value of the parameter
+	 * @param step step
+	 * @param name name of the observed value
+	 * @param description description
+	 * @param value observed value
+	 */
+	template <class T>
+	void observe(int64_t step, std::string name, std::string description, T value);
+
+	/**
+	 * Observe a registered tag.
+	 * @tparam T type of the tag
+	 * @param step step
+	 * @param name tag's name
+	 */
+	template <class T>
+	void observe_tag(int64_t step, std::string name);
```

Comment:
This can be used to observe registered tags instead. I used a different name so it is clear what it is doing.

Reply:
I think the name can be overloaded, i like doxygen docs for this and the interface is typed so all good...names for indicating types/behaviour is more python style imo ;)

Reply:
I think not having the value as argument kind of makes clear that something under the hood will be observed

Reply:
Update to observe().